client/recordUtil.js

```diff
@@ -301,13 +303,24 @@ module.exports.syncRecordAsJS = (record) => {
    * .toObject() options:
    * http://dcode.io/protobuf.js/global.html#ConversionOptions
    */
-  let object = record.toObject()
+  const object = api.SyncRecord.toObject(record)
   object.action = record.action
   const type = serializer.getSyncRecordObjectData(record)
+  if (!syncTypes[type]) {
+    throw new Error('Unknown sync objectData type: ' + type)
+  }
   object.objectData = type
-  const data = record[type].toObject({defaults: true, enums: Number, longs: Number})
+  const data = api.SyncRecord[syncTypes[type]].toObject(record[type], {
+    defaults: true,
+    enums: Number,
+    longs: Number
+  })
   if (data.fields && data.fields.length > 0) {
     object[type] = pickFields(data, data.fields)
+    // Empty the fields field for backwards compatibility with protobufjs 6.6
```

Comment:
@ayumi is this needed? the `fields` field seems to always be empty with protobufjs 6.6 but it is populated with the correct fields in 6.8. however, it is expected to be empty for the tests to pass.